gobblin-runtime/src/main/java/org/apache/gobblin/runtime/kafka/HighLevelConsumer.java

```diff
@@ -336,8 +336,8 @@ public QueueProcessor(BlockingQueue queue) {
     public void run() {
       log.info("Starting queue processing.. " + Thread.currentThread().getName());
       KafkaConsumerRecord record = null;
-      try {
-        while (true) {
+      while (true) {
+        try {
           record = queue.take();
```

Comment:
now an exception would cause us to skip processing the message (`record`), which doesn't seem like what we'd want, especially when using this class for kafka ETL.

shall we enclose only `processMessage` in a `try/catch`?

Reply:
This is happening even now, right? If an exception occurs we stop the processing, but we still continue consuming.

Reply:
right now the `QueueProcessor` thread is dead so we no longer run:
```
partitionOffsetsToCommit.put(partition, record.getOffset() + 1);
```
which means that whether or not we continue consuming, after system restart, we'll rewind offsets to have another go with that first problematic message

Reply:
But that is only if autoCommit is not enabled. We have our usecase with autoCommit enabled.